jgrapht-core/src/main/java/org/jgrapht/graph/interval/IntervalGraphMapping.java

```diff
@@ -0,0 +1,354 @@
+        for (int i = 1; i < vertices.size(); i++) {
+            V next = vertices.get(i);
+            if (current.getInterval().compareTo(next.getInterval()) > 0) {
+                isSorted = false;
+                break;
+            }
+            current = next;
+        }
+
+        if(isSorted) {
+            ArrayList<LinkedList<V>> edges = new ArrayList<>(vertices.size());
+            ArrayList<Interval<T>> vertexIntervals = new ArrayList<>(vertices.size());
+
+            // calculate all edges
+            for(int i = 0; i < vertices.size(); ++i) {
+                edges.add(i, new LinkedList<>());
+                for(int j = 0; j < i; ++j) {
+                    if(vertices.get(j).getInterval().getEnd().compareTo(vertices.get(i).getInterval().getStart()) >= 0) {
+                        edges.get(j).add(vertices.get(i));
+                    }
+                }
+            }
+
+            // add all vertices to the graph
+            for(int i = 0; i < vertices.size(); ++i) {
+                vertexIntervals.add(vertices.get(i).getInterval());
+                intervalMap.put(vertices.get(i).getInterval(), vertices.get(i));
+                graph.addVertex(vertices.get(i));
+            }
+
+            // build up the interval tree in linear time
+            this.intervalStructure = new IntervalTreeStructure<T>(vertexIntervals);
+
+            // add all calculated edges to the graph
+            for(int i = 0; i < vertices.size(); ++i) {
+                addIntervalEdges(vertices.get(i), edges.get(i));
+            }
+
+        } else { // if the list is not sorted, add vertices as usual. Thereby, the edges are calculated
+            for (V vertex : vertices) {
+                this.addVertex(vertex);
+            }
+        }
+    }
+
+    /**
+     * This method can only be used if the graph is an interval graph! Therefore, it is private. We use it for asIntervalGraphMapping()
+     *
+     * Construct a new live interval graph mapping with a new graph with given <code>vertices</code>. The graph can either be directed or undirected, depending on the
+     * specified edge factory.
+     *
+     * @param vertices initial vertices
+     * @param edges initial edges
+     * @param vertexSupplier the vertex supplier of the new graph.
+     * @param edgeSupplier the edge supplier of the new graph.
+     * @param weighted whether the graph is weighted, i.e. the edges support a weight attribute
+     *
+     * @throws NullPointerException if the specified edge factory is <code>
+     * null</code>.
+     */
+    private IntervalGraphMapping(ArrayList<V> vertices, Map<Pair<V, V>, E> edges, Supplier<V> vertexSupplier, Supplier<E> edgeSupplier, boolean weighted) {
+        this(new DefaultListenableGraph<>(new DefaultUndirectedGraph<>(vertexSupplier, edgeSupplier, weighted)));
+
+        ArrayList<Interval<T>> vertexIntervals = new ArrayList<>(vertices.size());
+
+        // add sorted vertices
```

Comment:
why are the vertices sorted? (I guess because of the call, but this comment looks quite confusing to me in this method)

Reply:
this method is private. I use it in asIntervalGraphMapping() and there I know that the list is sorted. I will edit the javaDoc so that it is clear that the list has to be sorted to be efficient.

Reply:
fixed javaDoc